kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/RowColumnReader.java

```diff
@@ -82,8 +79,7 @@ class RowColumnReader extends GroupConverter implements ParquetColumnReaders.Bas
               ? findSubFieldType(fileSchema, field, parquetFieldIdToTypeMap)
               : null;
       if (typeFromFile == null) {
-        if (StructField.METADATA_ROW_INDEX_COLUMN_NAME.equalsIgnoreCase(field.getName())
-            && field.isMetadataColumn()) {
+        if (field.getDataType().equals(MetadataType.ROW_INDEX)) {
```

Comment:
Interesting note:
Assume we add a new metadata column that the parquet reader is responsible for populating in the future.

If the parquet reader code isn't updated here, we'll build  a`NonExistentColumnReader` which I expect will eventually fail once we try to read null values with an unsupported `MetadataType`

This forces the reader to add handling for the new requested metadata column, and avoid silent bugs - in today's world, the parquet reader would happily return a column filled with nulls because it didn't find it in the data file.